### ISSUE TITLE
Server drops connection after approx 1 hour

### ISSUE THREAD

**Issue by runebakjacobsen** [tillsteinbach/VWsFriend#39]
After approximately an hour of runtime my server stops updating. I have attached the dump from the error, looks like a connection reset? 

```
vwsfriend_1   | INFO:VWsFriend:Updating data from WeConnect
grafana_1     | t=2021-09-09T09:10:20+0000 lvl=info msg="Request Completed" logger=context userId=0 orgId=0 uname= method=GET path=/ status=302 remote_addr=127.0.0.1 time_ms=0 size=29 referer=
vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 699, in urlopen
vwsfriend_1   |     httplib_response = self._make_request(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 445, in _make_request
vwsfriend_1   |     six.raise_from(e, None)
vwsfriend_1   |   File "<string>", line 3, in raise_from
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 440, in _make_request
vwsfriend_1   |     httplib_response = conn.getresponse()
vwsfriend_1   |   File "/usr/lib/python3.9/http/client.py", line 1345, in getresponse
vwsfriend_1   |     response.begin()
vwsfriend_1   |   File "/usr/lib/python3.9/http/client.py", line 307, in begin
vwsfriend_1   |     version, status, reason = self._read_status()
vwsfriend_1   |   File "/usr/lib/python3.9/http/client.py", line 268, in _read_status
vwsfriend_1   |     line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
vwsfriend_1   |   File "/usr/lib/python3.9/socket.py", line 704, in readinto
vwsfriend_1   |     return self._sock.recv_into(b)
vwsfriend_1   |   File "/usr/lib/python3.9/ssl.py", line 1241, in recv_into
vwsfriend_1   |     return self.read(nbytes, buffer)
vwsfriend_1   |   File "/usr/lib/python3.9/ssl.py", line 1099, in read
vwsfriend_1   |     return self._sslobj.read(len, buffer)
vwsfriend_1   | ConnectionResetError: [Errno 104] Connection reset by peer
vwsfriend_1   | 
vwsfriend_1   | During handling of the above exception, another exception occurred:
vwsfriend_1   | 
vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/requests/adapters.py", line 439, in send
vwsfriend_1   |     resp = conn.urlopen(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 755, in urlopen
vwsfriend_1   |     retries = retries.increment(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/util/retry.py", line 532, in increment
vwsfriend_1   |     raise six.reraise(type(error), error, _stacktrace)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/packages/six.py", line 769, in reraise
vwsfriend_1   |     raise value.with_traceback(tb)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 699, in urlopen
vwsfriend_1   |     httplib_response = self._make_request(
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 445, in _make_request
vwsfriend_1   |     six.raise_from(e, None)
vwsfriend_1   |   File "<string>", line 3, in raise_from
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/urllib3/connectionpool.py", line 440, in _make_request
vwsfriend_1   |     httplib_response = conn.getresponse()
vwsfriend_1   |   File "/usr/lib/python3.9/http/client.py", line 1345, in getresponse
vwsfriend_1   |     response.begin()
vwsfriend_1   |   File "/usr/lib/python3.9/http/client.py", line 307, in begin
vwsfriend_1   |     version, status, reason = self._read_status()
vwsfriend_1   |   File "/usr/lib/python3.9/http/client.py", line 268, in _read_status
vwsfriend_1   |     line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
vwsfriend_1   |   File "/usr/lib/python3.9/socket.py", line 704, in readinto
vwsfriend_1   |     return self._sock.recv_into(b)
vwsfriend_1   |   File "/usr/lib/python3.9/ssl.py", line 1241, in recv_into
vwsfriend_1   |     return self.read(nbytes, buffer)
vwsfriend_1   |   File "/usr/lib/python3.9/ssl.py", line 1099, in read
vwsfriend_1   |     return self._sslobj.read(len, buffer)
vwsfriend_1   | urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
vwsfriend_1   | 
vwsfriend_1   | During handling of the above exception, another exception occurred:
vwsfriend_1   | 
vwsfriend_1   | Traceback (most recent call last):
vwsfriend_1   |   File "/opt/venv/bin/vwsfriend", line 8, in <module>
vwsfriend_1   |     sys.exit(main())
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/vwsfriend/vwsfriend.py", line 194, in main
vwsfriend_1   |     weConnect.update(updateCapabilities=True, updatePictures=True, force=True)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/weconnect.py", line 439, in update
vwsfriend_1   |     self.updateVehicles(updateCapabilities=updateCapabilities, updatePictures=updatePictures, force=force)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/weconnect.py", line 484, in updateVehicles
vwsfriend_1   |     self.__vehicles[vin].update(fromDict=vehicleDict, updateCapabilities=updateCapabilities, updatePictures=updatePictures)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/elements/vehicle.py", line 173, in update
vwsfriend_1   |     self.updatePictures()
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/weconnect/elements/vehicle.py", line 408, in updatePictures
vwsfriend_1   |     imageDownloadResponse = self.weConnect.session.get(imageurl, stream=True)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/requests/sessions.py", line 555, in get
vwsfriend_1   |     return self.request('GET', url, **kwargs)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/requests/sessions.py", line 542, in request
vwsfriend_1   |     resp = self.send(prep, **send_kwargs)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/requests/sessions.py", line 655, in send
vwsfriend_1   |     r = adapter.send(request, **kwargs)
vwsfriend_1   |   File "/opt/venv/lib/python3.9/site-packages/requests/adapters.py", line 498, in send
vwsfriend_1   |     raise ConnectionError(err, request=request)
vwsfriend_1   | requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```

**Comment by tillsteinbach** [comment]
This looks like an unstable internet connection, but still it should not fail. I will try to fix that asap. Thanks for reporting!

**Comment by tillsteinbach** [comment]
Should be fixed in 0.4.3, if not please reopen the issue.
